src/msopenh264dec.cpp:67:11: error: no member named 'eOutputColorFormat' in 'TagSVCDecodingParam'
                        params.eOutputColorFormat = videoFormatI420;
                        ~~~~~~ ^
src/msopenh264enc.cpp:134:29: error: no member named 'sSliceCfg' in 'SSpatialLayerConfig'
                        params.sSpatialLayers[0].sSliceCfg.uiSliceMode = SM_DYN_SLICE;
                        ~~~~~~~~~~~~~~~~~~~~~~~~ ^
src/msopenh264enc.cpp:134:53: error: use of undeclared identifier 'SM_DYN_SLICE'
                        params.sSpatialLayers[0].sSliceCfg.uiSliceMode = SM_DYN_SLICE;
                                                                         ^
src/msopenh264enc.cpp:135:29: error: no member named 'sSliceCfg' in 'SSpatialLayerConfig'
                        params.sSpatialLayers[0].sSliceCfg.sSliceArgument.uiSliceSizeConstraint = ms_factory_get_payload_max_size(mFilter->factory);
                        ~~~~~~~~~~~~~~~~~~~~~~~~ ^

--- src/msopenh264dec.cpp.orig	2016-10-19 15:32:52 UTC
+++ src/msopenh264dec.cpp
@@ -62,7 +62,7 @@ void MSOpenH264Decoder::initialize()
 		mUnpacker=rfc3984_new();
 		if (mDecoder != 0) {
 			SDecodingParam params = { 0 };
-#if (OPENH264_MAJOR == 1) && (OPENH264_MINOR >=6)
+#if (OPENH264_MAJOR > 1) || ((OPENH264_MAJOR == 1) && (OPENH264_MINOR >=6))
 #else
 			params.eOutputColorFormat = videoFormatI420;
 #endif
--- src/msopenh264enc.cpp.orig	2017-07-20 13:56:32 UTC
+++ src/msopenh264enc.cpp
@@ -127,7 +127,7 @@ void MSOpenH264Encoder::initialize()
 			params.sSpatialLayers[0].iSpatialBitrate = targetBitrate;
 			params.sSpatialLayers[0].iMaxSpatialBitrate = maxBitrate;
 
-#if (OPENH264_MAJOR == 1) && (OPENH264_MINOR >=6)
+#if (OPENH264_MAJOR > 1) || ((OPENH264_MAJOR == 1) && (OPENH264_MINOR >=6))
 			params.sSpatialLayers[0].sSliceArgument.uiSliceMode = SM_SIZELIMITED_SLICE;
 			params.sSpatialLayers[0].sSliceArgument.uiSliceSizeConstraint = ms_factory_get_payload_max_size(mFilter->factory);
 #else
